src/protocol/requests/metadata/v4/request.js

```diff
@@ -12,6 +12,6 @@ module.exports = ({ topics, allowAutoTopicCreation = true }) => ({
   apiVersion: 4,
   apiName: 'Metadata',
   encode: async () => {
-    return new Encoder().writeArray(topics).writeBoolean(allowAutoTopicCreation)
+    return new Encoder().writeNullableArray(topics).writeBoolean(allowAutoTopicCreation)
```

Comment:
The protocol docs are saying: `If the topics array is null fetch metadata for all topics.`. It made sense to me to be specific here with the nullable array.